core/discovery.go

```diff
@@ -19,9 +19,48 @@ under the License.
 
 package core
 
-import "github.com/spf13/viper"
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
 
-// GetRootNode place holder function for providing a boostrap address for a Validating Peer.
-func GetRootNode() (string, error) {
-	return viper.GetString("peer.discovery.rootnode"), nil
+	"github.com/spf13/viper"
+
+	
+)
+
+// StaticDiscovery is an implementation of Discovery
+type StaticDiscovery struct {
+	rootNodes   []string
+	random      *rand.Rand
+	isValidator bool
+}
+
+// NewStaticDiscovery is a constructor of a Discovery implementation
+func NewStaticDiscovery(isValidator bool) *StaticDiscovery {
+	staticDisc := StaticDiscovery{}
+	staticDisc.rootNodes = strings.Split(viper.GetString("peer.discovery.rootnode"), ",")
+	staticDisc.random = rand.New(rand.NewSource(time.Now().Unix()))
+	staticDisc.isValidator = isValidator
+	return &staticDisc
+}
+
+// GetRootNode returns the first root node defined in peer.discovery.rootnode config
+// for validating peers, and a random one for non validating peers.
+func (sd *StaticDiscovery) GetRootNode() (string, error) {
+	if sd.isValidator {
+		return sd.rootNodes[0], nil
+	}
+	return sd.rootNodes[sd.random.Int()%(len(sd.rootNodes))], nil
+}
+
+// GetRootNodes returns a comma separated array of all the nodes defined in peer.discovery.rootnode config
+func (sd *StaticDiscovery) GetRootNodes() ([]string, error) {
+	var nodes []string = make([]string, len(sd.rootNodes))
+	if copied := copy(nodes, sd.rootNodes); copied != len(sd.rootNodes) {
```

Comment:
how about using:
`return append([]string(nil), sd.rootNodes...), nil`?
In that case, the error return can be removed as well.